src/pudl/scripts/dbt_helper.py

```diff
@@ -365,13 +366,37 @@ def update_row_counts(
 
     partition_expr = partition_expressions[0]  # TODO: support multiple partitions
     new = _calculate_row_counts(table_name, partition_expr)
-    combined = _combine_row_counts(filtered, new)
-    _write_row_counts(combined)
 
-    return UpdateResult(
-        success=True,
-        message=f"Successfully updated row count table with counts from {table_name}, partitioned by {partition_expr}.",
+    # Make old and new row counts comparable so we can detect changes
+    row_count_idx = ["table_name", "partition"]
+    std_old = (
+        old.astype("string")
+        .fillna(value="")
+        .sort_values(row_count_idx)
+        .reset_index(drop=True)
     )
+    std_new = (
+        new.astype("string")
+        .assign(partition=lambda df: df["partition"].replace("None", pd.NA))
+        .fillna(value="")
+        .sort_values(row_count_idx)
+        .reset_index(drop=True)
+    )
+
+    try:
+        pd.testing.assert_frame_equal(std_old, std_new, check_like=True)
+        return UpdateResult(
+            success=True,
+            message=f"Row counts for {table_name} are unchanged.",
+        )
```

Comment:
* Differentiating between no-op success and actual update success is helpful for identifying where changes are happening for debugging purposes.
* Is there any potential failure mode here by not attempting to update the file when we see there is no change?  Previously we would try to update it no matter what.